datastore.go

```diff
@@ -130,6 +131,31 @@ func DiskUsage(d Datastore) (uint64, error) {
 	return persDs.DiskUsage()
 }
 
+// TTLDatastore is an interface that should be implemented by datastores that
+// support expiring entries.
+type TTLDatastore interface {
+	Datastore
+
+	PutWithTTL(key Key, value interface{}, ttl time.Duration) error
+	SetTTL(key Key, ttl time.Duration) error
+}
+
+// Txn is an interface for transactions that can be committed or rolled back.
+type Txn interface {
```

Comment:
Maybe just `Tx` since the datastore is `TxDatastore`?

Reply:
went the other way, but made them consistent!